message/validation/consensus_validation.go

```diff
@@ -317,10 +317,17 @@ func (mv *messageValidator) validateBeaconDuty(
 	return nil
 }
 
-func (mv *messageValidator) hasFullData(signedMsg *specqbft.SignedMessage) bool {
+func (mv *messageValidator) needDuplicatedFullDataCheck(signedMsg *specqbft.SignedMessage) bool {
+	return (signedMsg.Message.MsgType == specqbft.ProposalMsgType ||
+		mv.isDecidedMessage(signedMsg)) &&
+		len(signedMsg.FullData) != 0
```

Comment:
I don't think we should have a case that fullData is empty for such message types

Reply:
Should I create a new rule that `fullData` must not be empty for proposal and decided and check it before full data hash check?

Reply:
if full data is missing for such msg types that also is an invalid message imo

Reply:
I agree. Fixed